vitest.config.ts

```diff
@@ -0,0 +1,18 @@
+import { join, relative, sep } from 'node:path';
+import { defineConfig } from 'vitest/config';
+
+export default defineConfig({
+  test: {
+    setupFiles: ['./test-setup.ts'],
+    resolveSnapshotPath(testPath, snapshotExtension) {
+      const p = relative(__dirname, testPath);
+      const parts = p.split(sep);
+      parts[0] = 'snapshots';
+      parts[parts.length - 1] = parts[parts.length - 1].replace(
+        '.ts',
+        snapshotExtension,
+      );
+      return join(__dirname, ...parts);
+    },
```

Comment:
Copied pretty much exactly from `snapshotResolver.cjs`, open to feedback regardless

Reply:
Is there a reason we don't let it be a default directory?

Reply:
it didn't work :D

Reply:
Can you elaborate?
Could it be it didn't work with `jest` but works with `vitest`?

Reply:
Sure, there were a lot of problems trying to make jest properly work with esnext modules work with typescript

Reply:
@amitdahan Can you try to remove all of the excessive code and look what will vitest do by default?

Reply:
See https://github.com/neondatabase/neonctl/pull/235#discussion_r1668048249

Specifically:

> Previously (the Jest setup) we ran tests from `dist/**/*.test.js` (which is also PART of the reason we had to `build` before `test`, but still need this because we are running the built CLI still).
> 
> This also explains why we persisted the snapshots manually in root `snapshots` dir!

I'll go ahead and remove all of this <picture data-single-emoji=":10_10:" title=":10_10:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T026T3BRN0P/10_10/5c80082528544406.png" alt=":10_10:" align="absmiddle"></picture>